### PR TITLE
Graph: Refactor graph utilities

### PR DESCRIPTION
**Pull request by orhankislal**
There are some functions (such as validation, generating grouping based query
strings etc.) that are duplicated in various files. This commit consolidates
them into generic utility functions.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/194/

File: src/ports/postgres/modules/graph/pagerank.py_in [hunk 1]
```diff
@@ -30,6 +30,8 @@
 import plpy
 from utilities.control import MinWarning
 from utilities.utilities import _assert
+from utilities.utilities import _check_groups
+from utilities.utilities import _grp_from_table
```
Comment (jingyimei) [inline src/ports/postgres/modules/graph/pagerank.py_in]: Currently, these functions are not used in pagerank, but similar functionalities are implemented. We should refactor those pagerank-customized implementations to use these functions. For instance the variables `group_by_clause` and `grouping_where_clause` can be created using the above functions.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/195/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/202/

**Pull request closed without merging**